launch/launch/launch_service.py

```diff
@@ -201,20 +203,21 @@ def _on_sigint(signum):
                 else:
                     self.__logger.warning('{} again, ignoring...'.format(base_msg))
 
-            def _on_sigterm(signum):
+            def _on_sigterm(signum: int) -> None:
                 signame = signal.Signals(signum).name
                 self.__logger.error(
                     'user interrupted with ctrl-\\ ({}), terminating...'.format(signame))
                 # TODO(wjwwood): try to terminate running subprocesses before exiting.
                 self.__logger.error('using {} can result in orphaned processes'.format(signame))
                 self.__logger.error('make sure no processes launched are still running')
-                this_loop.call_soon(this_task.cancel)
+                if this_task:
+                    this_loop.call_soon(this_task.cancel)
 
             with AsyncSafeSignalManager(this_loop) as manager:
                 # Setup signal handlers
                 manager.handle(signal.SIGINT, _on_sigint)
                 manager.handle(signal.SIGTERM, _on_sigterm)
-                if platform.system() != 'Windows':
+                if sys.platform != 'win32':
```

Comment:
Can you explain why this change is necessary? Why can't we keep using `platform.system()`?